### ISSUE TITLE
synapse 0.99 is slower than 0.34

### ISSUE THREAD

**Issue by richvdh** [matrix-org/synapse#4713]
subjectively, performance seems to be worse since 0.99. It might be just the .well-known stuff, but I'm not sure

**Comment by richvdh** [comment]
I rolled my server back to 0.34.1, and certainly it felt faster, subjectively. Unfortunately a bunch of other stuff didn't work well, and in any case it's hard to make objective comparisons since a large amount of code has changed.

I suspect the reason for the discrepancy is .well-known fetching. To confirm, I have disabled .well-known support on my server. Will monitor for a while.

**Comment by richvdh** [comment]
I had to re-enable .well-known support so that I could talk to people without SRV records. I can't say that I noticed my homeserver performing much better during the time I had it disabled.

However, one thing I *have* noticed is that .well-known support significantly increases the memory usage of my homeserver. Typically my HS currently stablises at a usage of around 2G, and this dropped to 1.5G with .well-known support disabled. The .well-known cache sits at around 1500 entries, so I can't believe it's all coming from there - it would imply about 333K per entry, which seems immense. I wonder if the connection pool is hanging on to data it shouldn't. It would probably be fruitful to investigate with a memory profiler.
